ebpf/c/tracer-ebpf.c

```diff
@@ -451,19 +451,16 @@ static void update_conn_stats(
     t.sport = ntohs(t.sport); // Making ports human-readable
     t.dport = ntohs(t.dport);
 
+    // initialize-if-no-exist the connection stat, and load it
+    conn_stats_ts_t empty = {};
+    bpf_map_update_elem(&conn_stats, &t, &empty, BPF_NOEXIST);
```

Comment:
Isn't this less efficient for cases we're we are just incrementing (which should be the majority of cases where this is called)? We'll be doing two lookups.

Reply:
Yes, but I don't know of a better way to do this.

Reply:
Whats wrong with the previous method? Were we hitting race conditions?

Reply:
There's a race condition with the following sequence:

- thread 1 handles a tcp_send event (a), gets an empty entry
- thread 2 handles a tcp_send event (b), gets an empty entry
 - thread one decides to pupulate the map for the first time, with sent = a.sent
- thread two decides to populate the map for the first time with sent = b.sent


Really we want the entry to be populated with a.sent + b.sent

Reply:
Good to know - were you able to reproduce that bug in tests?

Reply:
I was not. but nothing in the docs indicate that map access is safe across cores so I think this is worth adding.